Remove Preprocessor Checks for Old LLVM Bugs

LumioseSil opened the pull request:
#### 84250783c571255515d5c97a061c71620cf38570
<pre>
Remove Preprocessor Checks for Old LLVM Bugs
<a href="https://bugs.webkit.org/show_bug.cgi?id=196788">https://bugs.webkit.org/show_bug.cgi?id=196788</a>
&lt;rdar://problem/49792205&gt;

Reviewed by NOBODY (OOPS!).

Given how LLVM and clang have had their respective bugs resolved and
fixed, we do not need these workarounds anymore.

*Source\WTF\wtf\Int128.cpp: Remove clang + SSE3 workaround
*Source\ThirdParty\libwebrtc\Source\third_party\abseil-cpp\absl\numeric\int128.cc: Remove workaround
*Source\ThirdParty\libwebrtc\Source\third_party\abseil-cpp\absl\strings\internal\str_format\float_conversion.cc:
Remove workaround
*Source\ThirdParty\libwebrtc\Source\webrtc\rtc_base\logging.cc: Support
CFSTR macro that clang understands now
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/84250783c571255515d5c97a061c71620cf38570

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107927 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16983 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40811 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/117060 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/116408 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18372 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8300 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/100114 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113689 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/13866 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/97088 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/41686 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/95778 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/28725 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83429 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/1/builds/97184 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9907 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/30073 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/38/builds/96559 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/83/builds/7982 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10614 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/6965 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/38/builds/96559 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16059 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/49662 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 jsc-mips~~](https://ews-build.webkit.org/#/builders/37/builds/105564 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12186 "Built successfully") | | [  ~~🧪 jsc-mips-tests~~](https://ews-build.webkit.org/#/builders/37/builds/105564 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

Constellation reviewed (changes requested):
I think, if the author is not checking this, then this is not OK to remove this.

brentfulgham commented on Source/JavaScriptCore/runtime/CachedTypes.cpp:
```diff
@@ -440,11 +440,6 @@ class VariableLengthObject : public CachedObject<Source>, VariableLengthObjectBa
     template<typename T>
-#if CPU(ARM64) && CPU(ADDRESS32)
-    // FIXME: Remove this once it's no longer needed and LLVM doesn't miscompile us:
-    // <rdar://problem/49792205>
```
It looks like this was fixed back in 2019, so should be safe to remove.

Constellation replied:
Is it actually fixed? I think it is closed without much confirmation, so we do not know.

brentfulgham replied:
Oh, I see. The radar was closed under the belief that upstream LLVM changes taken around the time fixed it, but no definitive commit for the fix.

If you are uncomfortable changing this bit, we should just keep it.

AZero13 replied:
See the list of commits that are relevant to llvm clang that have been done over the years.

Constellation replied:
I'm unconfortable with this removal unless (1) the author ensured that this is fixed by testing or (2) the author points out the exact fix in LLVM and explain why this is already fixed. Just saying "there are many commits so it is probably already fixed" is not acceptable.

AZero13 replied:
https://github.com/llvm/llvm-project/commit/f1c28929125400a1680868f7c6eea720de256779

brentfulgham commented on Source/ThirdParty/libwebrtc/Source/third_party/abseil-cpp/absl/numeric/int128.cc:
```diff
@@ -111,27 +111,6 @@ uint128 MakeUint128FromFloat(T v) {
 }
 
-#if defined(__clang__) && !defined(__SSE3__)
-// Workaround for clang bug: https://bugs.llvm.org/show_bug.cgi?id=38289
```
This LLVM issue was fixed in April, 2019, so is likely safe to remove.

brentfulgham commented on Source/ThirdParty/libwebrtc/Source/third_party/abseil-cpp/absl/strings/internal/str_format/float_conversion.cc:
```diff
@@ -1079,14 +1079,7 @@ void PrintExponent(int exp, char e, Buffer *out) {
-      (!std::is_same<Float, long double>::value ||
-       !std::is_same<Int, uint64_t>::value) &&
-#endif
-      std::numeric_limits<Float>::digits <= std::numeric_limits<Int>::digits;
```
Ditto

Constellation reviewed (changes requested):
r- based on my comment.

brentfulgham commented on Source/ThirdParty/libwebrtc/Source/webrtc/rtc_base/logging.cc:
```diff
@@ -376,8 +376,7 @@ void LogMessage::OutputToDebug(LoggingSeverity severity, const LogLineRef& log_l
   // a preference to do so.
-  CFStringRef key = CFStringCreateWithCString(
-      kCFAllocatorDefault, "logToStdErr", kCFStringEncodingUTF8);
+  CFStringRef key = CFSTR("logToStdErr");
```
These two calls aren't precisely the same, but this is possibly a memory improvement.

brentfulgham commented on Source/WTF/wtf/Int128.cpp:
```diff
@@ -107,27 +107,6 @@ static UInt128Impl MakeUInt128FromFloat(T v) {
   return MakeUInt128(0, static_cast<uint64_t>(v));
 }
 
-#if COMPILER(CLANG) && !defined(__SSE3__)
```
Ditto -- this should be fixed in all current build environments.

brentfulgham reviewed (commented):
I think these changes are fine, but I think @youennf should confirm the WebRTC bits are reasonable.

geoffreygaren reviewed (changes requested):
r- based on discussion above

LumioseSil committed:
Remove Preprocessor Checks for Old LLVM Bugs
https://bugs.webkit.org/show_bug.cgi?id=196788
<rdar://problem/49792205>

Reviewed by NOBODY (OOPS!).

Given how LLVM and clang have had their respective bugs resolved and
fixed, we do not need these workarounds anymore.

*Source\WTF\wtf\Int128.cpp: Remove clang + SSE3 workaround
*Source\ThirdParty\libwebrtc\Source\third_party\abseil-cpp\absl\numeric\int128.cc: Remove workaround
*Source\ThirdParty\libwebrtc\Source\third_party\abseil-cpp\absl\strings\internal\str_format\float_conversion.cc:
Remove workaround
*Source\ThirdParty\libwebrtc\Source\webrtc\rtc_base\logging.cc: Support
CFSTR macro that clang understands now
